docs/orchestration.asciidoc

```diff
@@ -0,0 +1,106 @@
+* When an existing NodeSet is renamed, ECK performs the creation of a new NodeSet with the new name, and the removal of the old NodeSet, according to the NodeSet creation and removal patterns described above. Elasticsearch data is migrated away from the deprecated NodeSet before removal. The Elasticsearch resource <<{p}-update-strategy>> controls how many nodes can exist above or below the target node count during the upgrade.
+
+In every case, ECK ensures any StatefulSet operation is done in a way that is consistent with Elasticsearch orchestration best practices, by manipulating orchestration settings such as `discovery.seed_hosts`, `cluster.initial_master_nodes`, `discovery.zen.minimum_master_nodes`, and `_cluster/voting_config_exclusions` as necessary.
+
+[id="{p}-orchestration-limitations"]
+==== Limitations
+
+Due to the way Kubernetes and StatefulSets behave, ECK orchestration has the following limitations:
+
+* Storage requirements (including volume size) of an existing NodeSet cannot be updated. link:https://github.com/kubernetes/enhancements/issues/661[StatefulSet volumes expansion is not available yet in Kubernetes]. In order to upgrade the storage size, it is however possible to create a new NodeSet, or rename the existing one. Renaming a NodeSet will lead to the creation of a new StatefulSet with the specified storage size. The original StatefulSet is removed once Elasticsearch data has been migrated away to the nodes of the new StatefulSet.
+* Cluster availability cannot be guaranteed during the rolling upgrade of a one-node cluster, or for a cluster having indices with no replicas. If an Elasticsearch node holds the only copy of a shard, this shard becomes unavailable while the node is upgraded. Clusters with more than one node and at least one replica per index are considered best practice.
+* Elasticsearch Pods may stay `Pending` during a rolling upgrade if the Kubernetes scheduler cannot re-schedule them back. This is especially important when using local PersistentVolumes. If the Kubernetes node bound to a local PersistentVolume does not have enough capacity to host an upgraded Pod which was temporarily removed, that Pod will stay Pending.
+* Rolling upgrades can only make progress if the Elasticsearch cluster health is green. It is risky to attempt upgrading a cluster in the yellow state as some shards could become completely unavailable and degrade the cluster health to red. ECK takes the cautionary approach of waiting for green health before progressing but advanced users may force an upgrade by manually deleting pods themselves. The deleted pods will be automatically recreated at the latest revision. In the exceptional case that all the nodes of a NodeSet are unavailable, probably caused by a misconfiguration, the operator will ignore the cluster health and continue with the upgrade anyway.
```

Comment:
This sentence is a bit confusing:

The deleted pods will be automatically recreated at the latest revision. In the exceptional case that all the nodes of a NodeSet are unavailable, probably caused by a misconfiguration, the operator will ignore the cluster health and continue with the upgrade anyway.

I tried to reword it, but I am not sure if it's correct:

> The deleted pods are automatically recreated at the latest revision. If all the nodes of a NodeSet become unavailable, the operator ignores the cluster health and continues the upgrade.